docs/monitors/jenkins.md

```diff
@@ -0,0 +1,160 @@
+
+## Overview
+
+This monitor collects metrics from the Jenkins Metrics Plugin by hitting these endpoints:
+[../api/json](https://wiki.jenkins.io/display/jenkins/remote+access+api)
+(job metrics)  and
+[metrics/&lt;MetricsKey&gt;/..](https://wiki.jenkins.io/display/JENKINS/Metrics+Plugin)
+(default and optional Codahale/Dropwizard JVM metrics).
+
+Requires Jenkins 1.580.3 or later, as well as the Jenkins Metrics Plugin (see Setup).
+
+<!--- SETUP --->
+## Install Jenkins Metrics Plugin
+This monitor requires the Metrics Plugin in Jenkins. Go to `Manage Jenkins -> Manage Plugins -> Available -> Search "Metrics Plugin"`
+to find and install this plugin in your Jenkins UI.
+
+
+<!--- SETUP --->
+## Example Config
+
+Sample YAML configuration:
+
+```yaml
+monitors:
+- type: collectd/jenkins
+  host: 127.0.0.1
+  port: 8080
+  metricsKey: reallylongmetricskey
+```
+
+
+## Configuration
+
+To activate this monitor in the Smart Agent, add the following to your
+agent config:
+
+```
+monitors:  # All monitor config goes under this key
+ - type: jenkins
+   ...  # Additional config
+```
+
+**For a list of monitor options that are common to all monitors, see [Common
+Configuration](../monitor-config.md#common-configuration).**
+
+
+| Config option | Required | Type | Description |
+| --- | --- | --- | --- |
+| `httpTimeout` | no | `int64` | HTTP timeout duration for both read and writes. This should be a duration string that is accepted by https://golang.org/pkg/time/#ParseDuration (**default:** `10s`) |
+| `username` | no | `string` | Basic Auth username to use on each request, if any. |
+| `password` | no | `string` | Basic Auth password to use on each request, if any. |
+| `useHTTPS` | no | `bool` | If true, the agent will connect to the exporter using HTTPS instead of plain HTTP. (**default:** `false`) |
+| `httpHeaders` | no | `map of strings` | A map of key=message-header and value=header-value. Comma separated multiple values for the same message-header is supported. |
+| `skipVerify` | no | `bool` | If useHTTPS is true and this option is also true, the exporter's TLS cert will not be verified. (**default:** `false`) |
+| `caCertPath` | no | `string` | Path to the CA cert that has signed the TLS cert, unnecessary if `skipVerify` is set to false. |
+| `clientCertPath` | no | `string` | Path to the client TLS cert to use for TLS required connections |
+| `clientKeyPath` | no | `string` | Path to the client TLS key to use for TLS required connections |
+| `host` | **yes** | `string` | Host of the jenkins instance |
+| `port` | **yes** | `string` | Port of the jenkins instance |
+| `metricsKey` | **yes** | `string` | Api key to authenticate to the jenkins metrics plugin |
+| `jenkins_cluster` | **yes** | `string` | Dimension with a value that uniquely identifies the jenkins cluster |
+| `dimUpdateInterval` | no | `int64` | The interval (in minutes) at which to send a dimension update (default is 5 minutes). (**default:** `5m`) |
```

Comment:
This is unnecessary, the agent writer will handle deduping so just send the updates every interval that you send metrics.

Reply:
@keitwb 
While discussing this yesterday, there were two concerns doing this with regular interval:
- frequently calling jenkins unnecessarily to fetch the labels
- the worry of the channel buffer filling up in case of throttling (@jaysfx  has more context), not sure if this is still an issue if we retry frequently

Reply:
change is made